x-pack/plugins/spaces/public/management/edit_space/roles/component/space_assigned_roles_table.tsx

```diff
@@ -0,0 +1,494 @@
+    {
+      field: 'name',
+      name: i18n.translate('xpack.spaces.management.spaceDetails.rolesTable.column.name.title', {
+        defaultMessage: 'Role',
+      }),
+    },
+    {
+      field: 'privileges',
+      name: i18n.translate(
+        'xpack.spaces.management.spaceDetails.rolesTable.column.privileges.title',
+        {
+          defaultMessage: 'Privileges',
+        }
+      ),
+      render: (_, record) => {
+        const uniquePrivilege = new Set(
+          record.kibana.reduce((privilegeBaseTuple, kibanaPrivilege) => {
+            if (
+              kibanaPrivilege.spaces.includes(currentSpace.id) ||
+              kibanaPrivilege.spaces.includes('*')
+            ) {
+              if (!kibanaPrivilege.base.length) {
+                privilegeBaseTuple.push(
+                  i18n.translate(
+                    'xpack.spaces.management.spaceDetails.rolesTable.column.privileges.customPrivilege',
+                    {
+                      defaultMessage: 'custom',
+                    }
+                  )
+                );
+              } else {
+                return privilegeBaseTuple.concat(kibanaPrivilege.base);
+              }
+            }
+
+            return privilegeBaseTuple;
+          }, [] as string[])
+        );
+
+        return Array.from(uniquePrivilege).join(',');
+      },
+    },
+    {
+      field: 'metadata',
+      name: i18n.translate(
+        'xpack.spaces.management.spaceDetails.rolesTable.column.roleType.title',
+        {
+          defaultMessage: 'Role type',
+        }
+      ),
+      render: (_value: Role['metadata']) => {
+        return React.createElement(EuiBadge, {
+          children: _value?._reserved
+            ? i18n.translate(
+                'xpack.spaces.management.spaceDetails.rolesTable.column.roleType.reserved',
+                {
+                  defaultMessage: 'Reserved',
+                }
+              )
+            : i18n.translate(
+                'xpack.spaces.management.spaceDetails.rolesTable.column.roleType.custom',
+                {
+                  defaultMessage: 'Custom',
+                }
+              ),
+          color: _value?._reserved ? undefined : 'success',
+        });
+      },
+    },
+  ];
+
+  if (!isReadOnly) {
+    columns.push({
+      name: 'Actions',
+      actions: [
+        {
+          type: 'icon',
+          icon: 'lock',
+          href: '#',
+          target: '_self',
+          name: i18n.translate(
+            'xpack.spaces.management.spaceDetails.rolesTable.column.actions.reservedIndictor.title',
+            {
+              defaultMessage: 'Reserved',
+            }
+          ),
+          description: i18n.translate(
+            'xpack.spaces.management.spaceDetails.rolesTable.column.actions.reservedIndictor.description',
+            {
+              defaultMessage: 'No action to perform, this role is reserved',
+            }
+          ),
```

Comment:
__nit:__ I see we display this message also for non-reserved roles, but those that grant access to "*"/All spaces. Should there be a different tooltip?

Reply:
Yes! Good catch :) Pushed 4027175885c